### PR TITLE
Geolocation

### PR DESCRIPTION
**Pull request by pdw207**
page

### CODE REVIEW

**Comment by ignu** [comment]
I've been wondering for a while why not search by geolocation by default?

I imagine wanting to see events in other parts of the country is probably not the most common use case.

File: src/components/EventLocationFilter/EventLocationFilter.js [hunk 1]
```diff
@@ -27,8 +28,21 @@ class EventLocationFilter extends Component {
     };
 
     this._handleDocumentClick = this.handleDocumentClick.bind(this);
+    this.getGeolocation();
   }
 
+  getGeolocation() {
+    if (window.navigator.geolocation) {
+      window.navigator.geolocation.getCurrentPosition((pos) => {
+        const url = 'http://maps.googleapis.com/maps/api/geocode/json?latlng='+pos.coords.latitude+','+pos.coords.longitude+'&sensor=true';
```
Comment (pfarnach) [inline src/components/EventLocationFilter/EventLocationFilter.js]: I think it's cleaner if URLs/API calls happen in a separate API file (`/api`)

File: src/components/EventLocationFilter/EventLocationFilter.js [hunk 2]
```diff
@@ -27,8 +28,21 @@ class EventLocationFilter extends Component {
...
+        axios.get(url).then((res)=> {
+          const zipCode = res.data.results[0].address_components.filter((x)=>x.types.includes("postal_code"))[0].short_name;
+          console.log(zipCode);
```
Comment (pfarnach) [inline src/components/EventLocationFilter/EventLocationFilter.js]: Can you remove this?

**Comment by pfarnach** [comment]
@ignu that makes sense to me -- if the user denies the site the ability to get the location then they have to input it manually anyway, no harm no foul

**Comment by pdw207** [comment]
@InciteDemocracy We do need a UI element to show what the current filter is without opening the location popup. I suggest we make one request using lat and long to the api, we update the event listing and then we make a request to google api to update the zipcode and preload in the form.

**Comment by InciteDemocracy** [comment]
Yes, the most basic way to indicate the location is set is to change the displayed text "set location" to the zip or city that is set.

File: src/components/EventLocationFilter/EventLocationFilter.js (hunk 1 above)
Comment (pdw207) [inline src/components/EventLocationFilter/EventLocationFilter.js]: Agreed will make this change.

**Comment by pdw207** [comment]
I'll follow up with test coverage in a follow up PR.

**Comment by pdw207** [comment]
@pfarnach Can you take another look?

File: src/components/EventLocationFilter/EventLocationFilter.js [hunk 3]
```diff
@@ -49,59 +52,98 @@ class EventLocationFilter extends Component {
...
 
-  handleOnChangeLocation(e) {
-    // If user has entered exactly 5 digits, check to see if they're valid
-    // this avoids user having to trigger input blur to have field validated
-    const location = e.target.value;
+  getGeolocation() {
+    if (window.navigator.geolocation) {
+      window.navigator.geolocation.getCurrentPosition((pos) => {
+        const path = `?latlng=${pos.coords.latitude},${pos.coords.longitude}&sensor=true`;
+        axios.get(`http://maps.googleapis.com/maps/api/geocode/json${path}`).then((res) => {
```
Comment (pfarnach) [inline src/components/EventLocationFilter/EventLocationFilter.js]: As a matter of separation concerns it seems like we should only be making API calls from an API/Service module and not use `axios` in a component file.

File: src/components/Events/Events.js [hunk 4]
```diff
@@ -50,16 +55,32 @@ class Events extends Component {
...
+      if (!window.navigator.geolocation) { resolve(); }
+      window.navigator.geolocation.getCurrentPosition((pos) => {
+        this.setState({
+          geoLocation: {
+            lat: pos.coords.latitude,
+            long: pos.coords.longitude,
+            maxDistance: distanceRange[defaultRangeIndex].value,
+          }
+        });
+        resolve();
```
Comment (pfarnach) [inline src/components/Events/Events.js]: Are you relying on the new state where this promised is used?  If so, you may want to call resolve as the 2nd argument to `setState()`

**Comment by pdw207** [comment]
@pfarnach Waiting on @KamillaKhabibrakhmanova response to see if we can just get the zip code from the backend and avoid having to incur costs for the google api call.

**Comment by KamillaKhabibrakhmanova** [comment]
@pdw207 @pfarnach looks like this is definitely possible, I'd just probably need to get a custom endpoint that returns data for that lat/long. And we wouldn't need to use the google api

**Comment by pdw207** [comment]
@KamillaKhabibrakhmanova Can you tell me where the attribute would be nested in the response that includes the event listing? For example it is in the root? What is the exact field name? What I can do is populate the zipcode field with a lat/long if no zipcode is returned. This way we can get this merged and as soon as you implement  we will update the zipcode properly.

**Comment by KamillaKhabibrakhmanova** [comment]
@pdw207 This is now supported in the API: https://github.com/ResistanceCalendar/resistance-calendar-api#cities-endpoit

**Comment by pdw207** [comment]
@pfarnach Let me if you need more time to review.

**Pull request merged**